components/identity/org.wso2.carbon.identity.certificateauthority/src/main/java/org/wso2/carbon/identity/certificateauthority/CAAdminService.java

```
@@ -0,0 +1,288 @@
+    }
+
+    /**
+     * Gets the CSRs for the tenant CA having the given status.
+     *
+     * @param status The status filter
+     * @return CSRs of the tenant CA which has the given status
+     * @throws CAException
+     */
+    public CSR[] listCSRsByStatus(String status) throws CAException {
+        try {
+            String tenantDomain = CarbonContext.getThreadLocalCarbonContext().getTenantDomain();
+            List<CSR> csrList = csrService.listCSRsByStatus(tenantDomain, status);
+            return csrList.toArray(new CSR[csrList.size()]);
+        } catch (CAException e) {
+            log.error("Error when listing CSRs with status " + status, e);
+            throw new CAException("Error when listing CSRs");
+        }
+    }
+
+    /**
+     * Get the CSR specified by the given serial number.
+     *
+     * @param serialNo The serial number of the CSR
+     * @return CSR with the given serial number
+     * @throws CAException
+     */
+    public CSR getCSR(String serialNo) throws CAException {
+        try {
+            String tenantDomain = CarbonContext.getThreadLocalCarbonContext().getTenantDomain();
+            return csrService.getCSR(serialNo, tenantDomain);
+        } catch (CAException e) {
+            log.error("Error retrieving the CSR with serial no:" + serialNo, e);
+            throw new CAException("Error when retrieving the CSR");
+        }
+    }
+
+    /**
+     * Reject CSR without signing.
+     *
+     * @param serialNo The serial number of the CSR to be rejected
+     * @throws CAException
+     */
+    public void rejectCSR(String serialNo) throws CAException {
+        try {
+            String tenantDomain = CarbonContext.getThreadLocalCarbonContext().getTenantDomain();
+            csrService.rejectCSR(serialNo, tenantDomain);
+        } catch (CAException e) {
+            log.error("Error when rejecting the CSR with serial no:" + serialNo, e);
+            throw new CAException("Error when rejecting the CSR");
```

The original exception is swallowed here. Please fix.